flepimop/gempyor_pkg/src/gempyor/parameters.py

```diff
@@ -1,31 +1,45 @@
-import numpy as np
-import pandas as pd
-import pyarrow as pa
-import pyarrow.parquet as pq
+"""
+Abstractions for interacting with the parameters configurations.
+
+This module contains abstractions for interacting with the parameters section of given
+config files. Namely it contains the `Parameters` class.
+"""
+
+__all__ = ["Parameters"]
+
+
 import copy
-import confuse
-from numpy import ndarray
-import logging
-from . import model_info, NPI, utils
 import datetime
-import xarray as xr
+import logging
 import os
 
-logger = logging.getLogger(__name__)
+import confuse
+import numpy as np
+from numpy import ndarray
+import pandas as pd
 
-# TODO: it should work like
-# import xarray as xr
-# xr.DataArray(p_draw, dims=["parameter", "date", "subpop"],
-#                 coords={"parameter":modinf.parameters.pnames,
-#                 "date":pd.date_range(modinf.ti, modinf.tf, freq="D"),
-#                 "subpop":modinf.subpop_struct.subpop_names}).sel(parameter="gamma_0").plot()
+from . import NPI, utils
 
 
-## TODO: ideally here path_prefix should not be used and all files loaded from modinf
+logger = logging.getLogger(__name__)
 
 
 class Parameters:
```

Comment:
Erg, in python if you define attributes here instead of within `__init__` (or other methods) they become associated with the class object rather than the instance which has bad results for mutable objects:
```
>>> class FooBar:
...     abc = {}
...
>>> first = FooBar()
>>> second = FooBar()
>>> first.abc["test"] = 1
>>> second.abc
{'test': 1}
```
So I generally stay away from defining them here (mutable or immutable).

Reply:
got it - so fields declared in the class context are class-wide rather than instance-specific. like i said, reaaaaal rusty on python.